JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -51,45 +61,88 @@ fun JetnewsApp(
-                        AppDrawer(
-                            currentRoute = currentRoute,
-                            navigateToHome = navigationActions.navigateToHome,
-                            navigateToInterests = navigationActions.navigateToInterests,
-                            closeDrawer = { coroutineScope.launch { sizeAwareDrawerState.close() } }
-                        )
-                    },
-                    drawerState = sizeAwareDrawerState,
-                    // Only enable opening the drawer via gestures if we allow showing it
-                    gesturesEnabled = allowDrawerToBeShown
-                ) {
-                    JetnewsNavGraph(
-                        appContainer = appContainer,
-                        // Either allow showing the drawer, or show the nav rail
-                        showNavRail = !allowDrawerToBeShown,
-                        navController = navController,
-                        openDrawer = { coroutineScope.launch { sizeAwareDrawerState.open() } },
-                        navigationActions = navigationActions,
+            ModalDrawer(
+                drawerContent = {
+                    AppDrawer(
+                        currentRoute = currentRoute,
+                        navigateToHome = navigationActions.navigateToHome,
+                        navigateToInterests = navigationActions.navigateToInterests,
+                        closeDrawer = { coroutineScope.launch { sizeAwareDrawerState.close() } }
                     )
-                }
+                },
+                drawerState = sizeAwareDrawerState,
+                // Only enable opening the drawer via gestures if we allow showing it
+                gesturesEnabled = allowDrawerToBeShown
+            ) {
+                JetnewsNavGraph(
+                    appContainer = appContainer,
+                    // Either allow showing the drawer, or show the nav rail
+                    showNavRail = !allowDrawerToBeShown,
+                    navController = navController,
+                    openDrawer = { coroutineScope.launch { sizeAwareDrawerState.open() } },
+                    navigationActions = navigationActions,
+                )
             }
         }
     }
 }
 
+/**
+ * Returns the [WindowMetrics] corresponding to the current activities
+ * [WindowInfoRepository.currentWindowMetrics] as [State].
+ */
+@Composable
+private fun rememberCurrentWindowMetrics(): State<WindowMetrics> {
+    val activity = LocalContext.current.findActivity()
+    val currentWindowMetricsFlow = remember(activity) {
+        activity.windowInfoRepository().currentWindowMetrics
+    }
+    return currentWindowMetricsFlow.collectAsState(
+        WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(activity)
+    )
+}
+
+/**
+ * Remembers the [WindowSize] corresponding to the current window metrics.
+ */
+@Composable
+private fun rememberWindowSizeState(): WindowSize {
```

Comment:
Nit: As this is the API used above, move it above `rememberCurrentWindowMetrics(): State<WindowMetrics>`?

Reply:
I updated the order in `WindowSize.kt`, where these were moved to.